lib/susedistribution.pm

```diff
@@ -337,13 +337,22 @@ sub ensure_installed {
     my $pkglist = ref $pkgs eq 'ARRAY' ? join ' ', @$pkgs : $pkgs;
     $args{timeout} //= 90;
 
-    x11_start_program_xterm;
+    # In this context we want to call the x11_start_program from the distribution directly
+    $self->x11_start_program(default_gui_terminal());
+
     $self->become_root;
     ensure_serialdev_permissions;
     quit_packagekit;
     zypper_call "in $pkglist";
     wait_still_screen 1;
     send_key("alt-f4");    # close xterm
+
+    if (check_screen 'terminal-close-window', timeout => 30) {
+        wait_screen_change {
+            testapi::assert_and_click('terminal-close-window');
+        };
+    }
```

Comment:
I assume you're using `check_screen` here because only some terminals have a close prompt. Waiting 30 seconds in any other case seems rather wasteful.

I'm also wondering about the usefulness of the `wait_screen_change` call.

Reply:
without the `wait_screen_change` the close of the window was sometimes too slow and the next assert screen would fail (At least for gnome-console), same goes for the waiting time (doesn't take long, but at least missed once while developing the changes)

Reply:
How about `testapi::assert_and_click('terminal-close-window') if (check_screen 'terminal-close-window', timeout => 30)`